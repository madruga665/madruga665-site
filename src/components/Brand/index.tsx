import styles from "./styles.module.scss";

interface IPorps {
  withLogo?: boolean;
}

const Brand = ({ withLogo }: IPorps) => {
  return (
    <div className={styles.LogoName}>
      {withLogo || (
        <img src='/assets/icons8-circled-m-80.png' alt='Logo de Madruga' width={50} height={50} />
      )}
      <h2>
        {"<"}
        <span>L</span>
        uciano <span>A</span>
        mÃ¢ncio
        {"/>"}
      </h2>
    </div>
  );
};

export default Brand;
