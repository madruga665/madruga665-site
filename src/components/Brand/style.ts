import styled from "styled-components";
import { Colors } from "../../styles/variables";

export const LogoName = styled("div")`
  display: flex;
  align-items: center;

  h2 {
    font-size: 1.5rem;
    margin-left: 0.5rem;

    span {
      font-size: 1.8rem;
      color: ${Colors.emerald};
    }
  }
`;
