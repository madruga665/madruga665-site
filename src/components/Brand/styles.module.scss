@import '../../styles/colors.module';

.LogoName {
  display: flex;
  align-items: center;

  h2 {
    font-size: 1.5rem;
    margin-left: 0.5rem;

    span {
      font-size: 1.8rem;
      color: $color-emerald;
    }
  }
}