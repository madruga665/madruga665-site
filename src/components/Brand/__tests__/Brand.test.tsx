import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Brand from "..";

describe("Testes do componente Brand", () => {
  test("Snapshot do componente", () => {
    const tree = renderer.create(<Brand />).toJSON();

    expect(tree).toMatchSnapshot();
  });
  test("O logo esta sendo renderizado", () => {
    render(<Brand />);
    const madrugaLogo = screen.getByRole("img", {
      name: /logo de madruga/i,
    });
    expect(madrugaLogo).toBeInTheDocument();
  });
  test("O nome está sendo renderizado", () => {
    render(<Brand />);
    const madrugaName = screen.getByText(/<uciano mâncio\/>/i);
    expect(madrugaName).toBeInTheDocument();
  });
});
