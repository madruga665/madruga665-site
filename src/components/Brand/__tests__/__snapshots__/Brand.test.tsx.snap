// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testes do componente Brand Snapshot do componente 1`] = `
<div
  className="sc-bdvvtL hjtpYV"
>
  <img
    alt="Logo de Madruga"
    height={50}
    src="/assets/icons8-circled-m-80.png"
    width={50}
  />
  <h2>
    &lt;
    <span>
      L
    </span>
    uciano 
    <span>
      A
    </span>
    m√¢ncio
    /&gt;
  </h2>
</div>
`;
